Server/IT-Community.Server.Infrastructure/Services/CommentService.cs

```diff
@@ -0,0 +1,153 @@
+﻿using AutoMapper;
+using IT_Community.Server.Core.DataAccess;
+using IT_Community.Server.Core.Entities;
+using IT_Community.Server.Infrastructure.Dtos.CommentDTOs;
+using IT_Community.Server.Infrastructure.Dtos.PostDtos;
+using IT_Community.Server.Infrastructure.Exceptions;
+using IT_Community.Server.Infrastructure.Interfaces;
+using IT_Community.Server.Infrastructure.Resources;
+using Microsoft.AspNetCore.Identity;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IT_Community.Server.Infrastructure.Services
+{
+    public class CommentService : ICommentService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+        private readonly UserManager<User> _userManager;
+
+        public CommentService(IUnitOfWork unitOfWork, IMapper mapper, UserManager<User> userManager)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+            _userManager = userManager;
+        }
+
+        public List<CommentPostDto> GetCommentsWithReplies(int postId)
+        {
+            var commentList = _mapper.Map<List<CommentPostDto>>
+                (_unitOfWork.CommentRepository.GetAll(x => x.PostId == postId && x.ParentId == null, includeProperties: new[] {"Comments"}));
+            foreach(var comment in commentList)
+            {
+                LoadComments(comment, postId);
+            }
+
+            return commentList;
+        }
+
+        private void LoadComments(CommentPostDto comment, int postId)
+        {
+            foreach(var c in comment.ReplyList)
+            {
+                c.ReplyList = _mapper.Map<List<CommentPostDto>>(_unitOfWork.CommentRepository.GetAll(x => x.PostId == postId && x.ParentId == c.Id, includeProperties: new[] { "Comments" }));
+                foreach(var c2 in c.ReplyList)
+                {
+                    LoadComments(c2, postId);
+                }
+            }
+        }
+
+        private List<CommentPostDto> GetAllReplies(ref List<CommentPostDto> comments, int parentId)
+        {
+            List<CommentPostDto> innerComments = comments;
+
+            return comments.Where(x => x.ParentId == parentId).Select(m => new CommentPostDto()
+            {
+                Id = m.Id,
```

Comment:
Use auto mapper instead these lines.

Reply:
This method i don`t use, but forgot to delete